test/compress/screw-ie8.js

```diff
@@ -325,3 +325,61 @@ issue_2120_2: {
     }
     expect_stdout: "PASS"
 }
+
+issue_2254_1: {
+    mangle = {
+        ie8: false,
+    }
+    input: {
+        "eeeeee";
+        try {
+            console.log(f("PASS"));
+        } catch (e) {}
+        function f(s) {
+            try {
+                return s;
+            } catch (e) {}
+        }
+    }
+    expect: {
+        "eeeeee";
+        try {
+            console.log(f("PASS"));
+        } catch (e) {}
+        function f(e) {
+            try {
+                return e;
+            } catch (e) {}
+        }
+    }
+    expect_stdout: "PASS"
+}
+
+issue_2254_2: {
+    mangle = {
+        ie8: true,
+    }
+    input: {
+        "eeeeee";
+        try {
+            console.log(f("PASS"));
+        } catch (e) {}
+        function f(s) {
+            try {
+                return s;
+            } catch (e) {}
+        }
+    }
+    expect: {
+        "eeeeee";
+        try {
+            console.log(f("PASS"));
+        } catch (e) {}
+        function f(c) {
+            try {
+                return c;
+            } catch (e) {}
```

Comment:
While CLI can disable `compress`, you can't do that with `test/run-tests.js`
https://github.com/mishoo/UglifyJS2/blob/a5ffe2c23fdfaf13f3466a01d9dd9d590c5e8672/test/run-tests.js#L114-L115

So with your example:
```js
issue_2254_kzc: {
    mangle = {
        ie8: true,
    }
    input: {
        try {} catch (e) {}
        function eee(a) {
            try {
                b.what = a;
            } catch (e) {
                console.log(a);
            }
        }
        eee("PASS");
    }
    expect: {
        try {} catch (e) {}
        function eee(e) {
            try {
                b.what = e;
            } catch (e) {
                console.log(e);
            }
        }
        eee("PASS");
    }
}
```
`npm test` will fail with:
```
!!! failed
---INPUT---
{
    try {} catch (e) {}
    function eee(a) {
        try {
            b.what = a;
        } catch (e) {
            console.log(a);
        }
    }
    eee("PASS");
}
---OUTPUT---
function eee(e){try{b.what=e}catch(c){console.log(e)}}eee("PASS");
---EXPECTED---
try{}catch(e){}function eee(e){try{b.what=e}catch(e){console.log(e)}}eee("PASS");
```